src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComInterfaceMarshaller.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+
+namespace System.Runtime.InteropServices.Marshalling
+{
+    /// <summary>
+    /// COM interface marshaller using a StrategyBasedComWrappers instance
+    /// </summary>
+    /// <remarks>
+    /// This marshaller will always pass the <see cref="CreateObjectFlags.Unwrap"/> flag
+    /// to <see cref="ComWrappers.GetOrCreateObjectForComInstance(IntPtr, CreateObjectFlags)"/>.
+    /// </remarks>
+    [UnsupportedOSPlatform("android")]
+    [UnsupportedOSPlatform("browser")]
+    [UnsupportedOSPlatform("ios")]
+    [UnsupportedOSPlatform("tvos")]
+    [CLSCompliant(false)]
+    [CustomMarshaller(typeof(CustomMarshallerAttribute.GenericPlaceholder), MarshalMode.Default, typeof(ComInterfaceMarshaller<>))]
+    public static unsafe class ComInterfaceMarshaller<T>
+    {
+        public static void* ConvertToUnmanaged(T? managed)
+        {
+            if (managed == null)
+            {
+                return null;
+            }
+            if (!ComWrappers.TryGetComInstance(managed, out nint unknown))
+            {
+                unknown = StrategyBasedComWrappers.DefaultMarshallingInstance.GetOrCreateComInterfaceForObject(managed, CreateComInterfaceFlags.None);
+            }
```

Comment:
As of now, the `ComInterfaceMarshaller` will unwrap any RCW. Do we want to introduce a `CreateComInterfaceFlags.Unwrap` option to provide the same unwrapping functionality of `GetOrCreateObjectForComInstance` (unwrapping a wrapper created by this ComWrappers instance)?

Reply:
I don't understand that. There can only be a single identity associated with an RCW.